packages/flutter_tools/gradle/aar_init_script.gradle

```diff
@@ -22,7 +39,17 @@ void configureProject(Project project, String outputDir) {
     // to resolve dependencies.
     project.version = project.version.replace("-SNAPSHOT", "")
 
-    if (project.hasProperty("buildNumber")) {
+    if (!isModuleProject) {
+        def deps = pubspecLock.packages.find{ it.key == "${project.name}" }
+        if (deps != null && deps.value != null && deps.value.version != null && !(deps.value.version.trim().isEmpty())){
+            def _version = deps.value.version
+            project.version = _version
```

Comment:
Override default version number with the version number obtained from the YAML instance.